db/changes.go

```diff
@@ -171,6 +172,160 @@ func (db *Database) AddDocInstanceToChangeEntry(entry *ChangeEntry, doc *Documen
 	}
 }
 
+func (db *Database) buildRevokedFeed(singleChannelCache SingleChannelCache, options ChangesOptions, triggeredBy uint64) <-chan *ChangeEntry {
+	feed := make(chan *ChangeEntry, 1)
+	sinceVal := options.Since.Seq
+
+	queryLimit := db.Options.CacheOptions.ChannelQueryLimit
+	requestLimit := options.Limit
+
+	paginationOptions := options
+	paginationOptions.Since.Seq = 0
+	paginationOptions.Since.LowSeq = 0
+
+	if options.Since.TriggeredBy > 0 {
+		// Need to check if there has been a change since the triggeredByVal. If there has we need to start again from
+		// 0. Otherwise we can continue where we left off
+		highSeq, err := db.user.FindMostRecentRevocationSeq(singleChannelCache.ChannelName())
+		if highSeq > triggeredBy && err == nil {
+			paginationOptions.Since.Seq = options.Since.TriggeredBy
+		}
+	}
+
+	go func() {
+		defer base.FatalPanicHandler()
+		defer close(feed)
+		var itemsSent int
+		var lastSeq uint64
+
+		for {
+			if requestLimit == 0 {
+				paginationOptions.Limit = queryLimit
+			} else {
+				remainingLimit := requestLimit - itemsSent
+				paginationOptions.Limit = base.MinInt(remainingLimit, queryLimit)
+			}
+
+			// Get changes from 0 to latest seq
+			base.TracefCtx(db.Ctx, base.KeyChanges, "Querying channel %q for revocations with options: %+v", base.UD(singleChannelCache.ChannelName()), paginationOptions)
+			changes, err := singleChannelCache.GetChanges(paginationOptions)
+			if err != nil {
+				base.WarnfCtx(db.Ctx, "Error retrieving changes for channel %q: %v", base.UD(singleChannelCache.ChannelName()), err)
+				change := ChangeEntry{
+					Err: base.ErrChannelFeed,
+				}
+				feed <- &change
+				return
+			}
+			base.DebugfCtx(db.Ctx, base.KeyChanges, "[changesFeed] Found %d changes for channel %q", len(changes), base.UD(singleChannelCache.ChannelName()))
+
+			sentChanges := 0
+			for _, logEntry := range changes {
+				seqID := SequenceID{
+					Seq:         logEntry.Sequence,
+					TriggeredBy: triggeredBy,
+				}
+
+				idOnlyRevocation := false
+
+				// We need to check whether a change / document sequence is greater than since.
+				// If its less than we can continue. Otherwise:
+				// Because we only see the active revision of the document through the GetChanges query, we need to check
+				// whether the user had access to this document at the last since to know whether we need to  end a
+				// revocation message. We don't want to send a revocation message if the document has not seen before.
```

Comment:
I think the three lines above might be simplified as:
// Otherwise, we need to determine whether a previous revision of the document was in the 
// channel prior to the since value, and only send a revocation if this was the case.